diagnostics-app/src/app/diagnostics/diagnostics-card/diagnostics-card.component.html

```diff
@@ -0,0 +1,55 @@
+<!-- 
+ Copyright 2023 The Chromium Authors. All rights reserved.
+ Use of this source code is governed by a BSD-style license that can be
+ found in the LICENSE file.
+-->
+
+<div class="diagnostics-card">
+  <div class="header-bar">
+    <div class="header-and-params">
+      <header>{{ routine.toUpperCase() }}</header>
+      <div class="params-info" *ngIf="params">
+        <div class="param-info" *ngFor="let param of $any(params) | keyvalue: originalOrder">
+          {{ param.key }}: {{ param.value }}
+        </div>
+      </div>
+    </div>
+    <div class="buttons">
+      <button mat-raised-button color="basic" (click)="startRoutine()" [disabled]="state !== 0">Run</button>
+      <button mat-raised-button color="warn" (click)="stopRoutine()" [disabled]="state !== 1">Stop</button>
+      <button mat-raised-button color="basic" (click)="resumeRoutine()" [disabled]="!needResume() || state !== 2">Resume</button>
```

Comment:
Using numbers (0, 1, 2) is not readable.

Sorry for lack of insight, could we use enum here?

Reply:
A bug is filed for this: [b/302096289](url).